### PR TITLE
feat(common): appex-447 updates big-design/firebase/styled-components

### PR DESCRIPTION
**Pull request by bc-zachary**
## What?
Updates big-design, firebase and styled-components dependencies and updates snapshots for BD and styled-components.

## Why?
Snyk alerts

## Testing / Proof
Tests passing and verified running/build/lint locally

@bigcommerce/api-client-developers

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "@types/jest": "^27.5.2",
     "@types/node": "^18.0.0",
     "@types/react": "^17.0.2",
-    "@types/styled-components": "^5.1.25",
+    "@types/styled-components": "^5.1.26",
```
Comment (bc-chaz) [inline package.json:41]: while we're here, should we consider upgrading to the latest version? i.e. I believe we're on 5.3.6 - from what I can tell there aren't any breaking changes

File: package.json [hunk 2]
```diff
@@ -19,16 +19,16 @@
     "npm": ">=7.20.x <8.14"
   },
   "dependencies": {
-    "@bigcommerce/big-design": "^0.32.1",
+    "@bigcommerce/big-design": "^0.33.0",
```
Comment (bc-chaz) [inline package.json:22]: qq: does this break anything if we use the latest version of BD? `0.34.2`

File: package.json [hunk 3]
```diff
@@ -19,16 +19,16 @@
...
     "@bigcommerce/big-design-theme": "^0.16.0",
-    "firebase": "^9.9.2",
+    "firebase": "^9.9.3",
```
Comment (bc-chaz) [inline package.json:24]: we might want to give this a slight bump as well - there's a `9.9.4` that's probably pretty safe

File: package.json (hunk 2 above)
Comment (bc-zachary) [reply package.json:22]: Nope! I'll push that up. 👍

File: package.json (hunk 3 above)
Comment (bc-zachary) [reply package.json:24]: I think with the carat `^` preceding the version, `npm i` automatically downloads the latest minor update — which in this case is `9.13.0` — and it's building fine for me locally so I think I could even bump it to that unless we want to be able to use `9.9.x`

File: package.json [hunk 4]
```diff
@@ -19,16 +19,16 @@
...
     "jsonwebtoken": "^8.5.1",
     "mysql": "^2.18.1",
     "next": "^12.2.5",
```
Comment (matthewvolk) [inline package.json:27]: Maybe we should make a ticket to bump us to Next 13! Next.js Conf earlier this week has me like: 🤤😍

**Review by matthewvolk — COMMENTED** [review]
Awesome! Will approve as soon as those minor bumps mentioned above are pushed.

**Review by matthewvolk — APPROVED** [review]
LGTM 👍

**Pull request merged**